Improve localization guide (#341)
cncf/glossary: LOCALIZATION.md

Now that the first version of the cloud native glossary is live,
we'd love to start localizing it into different languages. 

 - [Initiate a new localization team](#initiate-a-new-localization-team)
 - [Get involved in an existing localization team](#get-involved-in-an-existing-localization-team)

Contributions to improving the following localization guides and localization policies are also welcome.

---

# Initiate a new localization team

## Basic requirements to initiate a new localization team

- The Glossary project does not have a localization team for your language yet
  - check the language list from [localization progress](https://github.com/cncf/glossary/projects/2) and ask in the [glossary-localizations channel in Slack](https://cloud-native.slack.com/archives/C02N2RGFXDF). There may be a group of volunteers who haven't gotten started yet.
- At least three volunteers
- At least two approvers 
  - two approvers must approve each term
  - unanimous consent must be reached to be approved 

>   - at least one volunteer must have basic GitHub knowledge and be willing to help the others get up to speed (such as PR, review process, branch, ..)

Reviewer: I think this should be at least one, not all. But this person should be willing to help others.

## Instructions to initiate a new localization

> ### 1. Opening an issue to notify initiation of a new localization 

Reviewer: We should also mention joining Slack. I think that may be the first step, what do you think?
****

# 1. Joining the Slack channel and saying hi

If you're interested in creating a new localization team, whether you have a team or are on your own, please join the #glossary-localizations channel on the CNCF slack and say hi. If you have at least three volunteers and are ready to start, request a localization channel for your language (e.g., #glossary-localization-korean). If you are still looking for more volunteers, let everyone in the channel know. There may be someone in there looking for a team as well.

[Open an issue](https://github.com/cncf/glossary/issues/new/choose) 
and select the `Initiate a New Localization Team` template.

With this issue, new localization teams are requesting:
 - their own development branch (ex: [dev-ko](https://github.com/cncf/glossary/tree/dev-ko) branch). 
 - permission for approving localized content
   ```
   l10n team approvers will have push permission to this repository.
   They will manage (merge) PRs for their l10n contributions in their development branch.
   However, their ability to merge PRs to the `main` branch is restricted.
   Please note: although they can review and approve PRs to the `main` branch, they should not approve PRs that aren't related to their own localization teams.
   ``` 

Localization teams should use their assigned development branch for the following localization tasks (example of a dev branch: https://github.com/cncf/glossary/tree/dev-ko; dev-ko stands for Korean dev branch). 

### 2. Configuring a new language

Open a PR with the localization initiation following this example: https://github.com/cncf/glossary/pull/291.

#### 2-1. Adding a new language setting to site configuration

> New languages need to be set up inside of `config.toml`.
> Find existing `[languages]` block in `config.toml` and add a block for the new language below other language blocks.

Author: ```suggestion
To add a new language to site, you needs to modify [config.toml](https://github.com/cncf/glossary/blob/main/config.toml#L54) (Note: localization teams should use their assigned development branch for this task).
`[languages]` block of `config.toml` is used to configure languages for the site. `[languages.en]` means language configuration for English and `[languages.ko]` means language configuration for Korean.
Go to `[languages]` block in `config.toml` and add a new block like `[languages.ko]` with a language-specific configuration.
For instance, Korean localization team can add `[languages.ko]` block after `[languages.en]` block.
```
How about this ? :)

Reviewer: Looks good. I made a few edits. Can you verify, it's all correct. If so, let's commit.

Now that the first version of the cloud native glossary is live,
we'd love to start localizing it into different languages. 

 - [Initiate a new localization team](#initiate-a-new-localization-team)
 - [Get involved in an existing localization team](#get-involved-in-an-existing-localization-team)

Contributions to improve following localization guides and localization policies are also welcome.

---

# Initiate a new localization team

## Basic requirements to initiate a new localization team

- Glossary project does not have a localization team for your language yet
  - check language list from [localization progress](https://github.com/cncf/glossary/projects/2) or ask it to [glossary-localizations channel in Slack](https://cloud-native.slack.com/archives/C02N2RGFXDF)
- At least three volunteers
- At least two approvers 
  - each term must be approved by two approvers
  - unanimous consent must be reached to be approved 

>   - candidates need to have basic knowledge in Github (such as PR, review process, branch, ..)

Author: Looks better :) Thanks!

## Instructions to initiate a new localization

### 1. Open an Issue to notify initiation of a new localization 

[Open an Issue](https://github.com/cncf/glossary/issues/new/choose) 
based on the `Initiate a New Localization Team` Issue-template.

Through this issue, the new localization team will
 - get a development branch for the localization works (ex: [dev-ko](https://github.com/cncf/glossary/tree/dev-ko) branch). 
 - get a permission for approving localization contents.
   ```
   The approvers of l10n team will have a push permission to this repository.
   It is to make l10n approvers manage (merge) PRs for l10n contributions in your development branch.
   Merging a PR to the `main` branch by l10n approvers is restricted.

>    Even if they are possible to review a PR to the `main` branch and give an approval to the PR, they should not approve ths PR. 

Author: ```suggestion
   Even if they are possible to review a PR to the `main` branch and give an approval to the PR, they should not approve the PR. 
```

   So, please do not approve if a PR is not related with your localization.
   ``` 

The localization team needs to use assigned development branch for following localization tasks.
For instance, Korean localization team is using [dev-ko](https://github.com/cncf/glossary/tree/dev-ko) branch.

### 2. Configure a new language to the site

Fulfill following tasks and open a PR. (ex: https://github.com/cncf/glossary/pull/291)

#### 2-1. Add a new language setting to site configuration

New languages need to be set up inside of `config.toml`.

> Find existing `[languages]` block in `config.toml` and add a block for the new language below other language blocks.

Author: https://github.com/cncf/glossary/pull/341#discussion_r790303902

- At least two approvers 
  - two approvers must approve each term
  - unanimous consent must be reached to be approved 
  - at least one volunteer must have basic GitHub knowledge and be willing to help the others get up to speed (such as PR, review process, branch, ..)

## Instructions to initiate a new localization

# 1. Join the Slack channel and saying hello

If you're interested in creating a new localization team, whether you have a team or are on your own, please join the #glossary-localizations channel on the CNCF slack and say hi. If you have at least three volunteers and are ready to start, request a localization channel for your language (e.g., #glossary-localization-korean). If you are still looking for more volunteers, let everyone in the channel know. There may be someone in there looking for a team as well. 

### 2. Opening an issue to notify initiation of a new localization 

[Open an issue](https://github.com/cncf/glossary/issues/new/choose) 
and select the `Initiate a New Localization Team` template.

With this issue, new localization teams are requesting:
 - their own development branch (ex: [dev-ko](https://github.com/cncf/glossary/tree/dev-ko) branch). 

>  - permission for approving localized content

Author: ```suggestion
 - permission for approving localized content
   - based on discussions, a l10n team needs to select suitable candidates for l10n content approvers (up to 4 candidates)
```
@CathPag 
How about adding this sentence to guide the maximum approvers for each team :)

Before opening a PR, ensure the website with the updated configuration works by running [Hugo server](https://github.com/cncf/glossary#setting-up-a-local-instance). If it does, select the new language on the local website.

![image](https://user-images.githubusercontent.com/5966944/150203331-c535a95c-c36e-4c01-85a8-5261ce6f70de.png)

Open a PR (ex: https://github.com/cncf/glossary/pull/291) and wait for it to be reviewed by the [maintainers](https://github.com/cncf/glossary/blob/main/CODEOWNERS#L6).

### 3. Localizing and adding minimum required terms

Before going live with a localized Glossary, you need at least:

 - 10 translated terms

How localization teams achieve that, is up to them. For instance, they can use issues to assign contributors to translation terms and track progress (ex: https://github.com/cncf/glossary/issues/269).

### 4. Merging the localization branch into `main` branch

After completing the minimum required terms for a new localization, open a PR to merge commits in the localization branch into the `main` branch. (ex: `dev-ko` to `main`).

Once the PR is merged, the localized content will go live on its website 🎉

---

> # Join an existing localization team

Reviewer: Updated this section. I don't think we need a lot more. If there is a team already, they should be able to onboard the newbie, what do you think?

Author: Looks good to me :)
